init-pants/action.yaml

```diff
@@ -1,62 +1,84 @@
+  Install Pants and its dependencies and prepare its caches.
 
 inputs:
-  # inputs are always string type
-  pants-python-version:
+  # Note: inputs are always string typed.
+  setup-commit:
+    description: |
+      Run the get-pants.sh script version at this commit in the pantsbuild/setup repo.
+      Picking a specific commit is safer than downloading the current version.
     required: false
-    default: '3.9'
+    # When we update get-pants.sh, we should update this commit.
+    default: baaf73b00d1f0c0508cfbdf4987dc9caa69d85b9
   base-branch:
+    description: |
+      The fallback commit to restore the local process cache from, if no cache
+      is available at the current commit.
     required: false
     default: main
   gha-cache-key:
-    required: true
-    # use this for matrix.python-version or similar
+    description: Qualify all cache keys with this string. Useful for invalidating everything.
+    required: false
+    default: v0
   named-caches-hash:
+    description: |
+      The hash to cache the named caches against. Should be a hash generated by hashFiles()
+      on all inputs to named caches, which are typically your lockfiles.
     required: true
-    # should be a hash generated by hashFiles()
   pants-ci-config:
+    description: |
+      An extra config file to use in CI. Defaults to pants.ci.toml if present, or no
+      file otherwise. Set to an empty string to disable.
     required: false
     default: DEFAULT
-    # defaults to pants.ci.toml if present
-    # set to empty string to disable
   cache-lmdb-store:
+    description: |
+      Whether to upload and restore the local process cache. If you're not using a fine-grained
+      remote caching service (see https://www.pantsbuild.org/docs/remote-caching), then you may
+      also want to preserve the local Pants process cache (lmdb_store). However this must
+      invalidate for changes to any file that can affect the build, so may not be practical in
+      larger repos. A remote cache service integrates with Pants's fine-grained invalidation and
+      avoids these problems.
     required: false
     default: 'false' # a string!
 
 runs:
   using: "composite"
   steps:
-    - name: 'Set up Python for Pants (${{ inputs.pants-python-version }})'
-      uses: actions/setup-python@v4
-      with:
-        python-version: '${{ inputs.pants-python-version }}'
-
-    - name: Get Pants bootstrap cache key
-      id: pants_version
+    - name: Ensure the Pants launcher binary
       shell: bash
       run: |
-        if ! grep 'PANTS_BOOTSTRAP_TOOLS' ./pants > /dev/null; then
-            echo "This action requires a newer ./pants script. Please update following https://www.pantsbuild.org/docs/installation"
-            exit 1
+        if ! command -v pants; then
+          curl --proto '=https' --tlsv1.2 -fsSLO \
+            https://raw.githubusercontent.com/pantsbuild/setup/${{ inputs.setup-commit }}/get-pants.sh
+          ./get-pants.sh
```

Comment:
How does `pants` get on the PATH? is ~/bin luckily on it in all GH Runner images? It would seem wise to take full control and use switches perhaps and explicitly add to the PATH in this action f possible. Then the action just works with any runner even self-hosted.

Reply:
For the record, I never use `~/bin`. I use `~/.local/bin`. I don't know if `~/bin` is on the PATH by default in GHA.

Reply:
Ooops, we should definitely add it to the PATH...

Reply:
~/bin is where get-pants.sh downloads to. GHA's example of "how to prepend to the PATH" does use `$HOME/.local/bin`, so maybe we should use that in get-pants.sh? I'm not sure how standard any of these things are.

Reply:
Completely non standard. We should probably force a choice or probe each PATH entry and order by writable dirs 1st and force a choice.

Reply:
I guess `$HOME/.local/bin` is the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) recommendation.

Reply:
Nack for Windows. We really ought not be cute. It is probably just like ICs and we should force a modicum of effort.

Reply:
Yeah. I've used `~/.local` for many years. I don't remember when or why I started using it, but the XDG spec was very likely part of my reasoning.

Reply:
+1 for `~/.local` on *nix systems.